app/views/posts/index.html.erb

```diff
@@ -1 +1,29 @@
-<h1>The posts for the users</h1>
+<div>
+  <%= render partial: "users/user_profile", locals: { user: @user } %>
+    <ul>
+      <% @posts.each do |p| %>
+        <li>
+          <div class="border border-3 border-dark container mt-5 p-2 bio">
+            <%= link_to user_post_path(@user.id, p.id), id: "user_post#{p.id}" do %>
```

Comment:
- [ ] [OPTIONAL] to make this slightly more succinct you don't even need to pass in the id for these models. rails will know how to navigate if you simply give it the model.
```suggestion
            <%= link_to user_post_path(@user, p), id: "user_post#{p.id}" do %>
```